File: capstoneSDETPython/pageObjects/LoginPage.py
from selenium import webdriver
from selenium.webdriver.common.by import By


class LoginPage:
    username_xpath = "//input[@id='user-name']"
    password_xpath = "//input[@id='password']"
    login_xpath = "//input[@id='login-button']"
    logo_xpath = "//div[@class='app_logo']"

    def __init__(self, driver):
        self.driver = driver

    def setUserName(self, username):
        self.driver.find_element(By.XPATH, self.username_xpath).clear()
        self.driver.find_element(By.XPATH, self.username_xpath).send_keys(username)

    def setPassword(self, password):
        self.driver.find_element(By.XPATH, self.password_xpath).clear()
        self.driver.find_element(By.XPATH, self.password_xpath).send_keys(password)

    def clickLogin(self):
        self.driver.find_element(By.XPATH, self.login_xpath).click()

    def getLogo(self):
        return self.driver.find_element(By.XPATH, self.logo_xpath).text

File: capstoneSDETJava/src/main/java/pageobjects/LoginPage.java
package pageobjects;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import testbase.BaseClass;

public class LoginPage extends BaseClass {

    By userNameXpath = By.xpath("//input[@id='user-name']");
    By passwordXpath = By.xpath("//input[@id='password']");
    By loginBtnXpath = By.xpath("//input[@id='login-button']");
    By loginErrorXpath = By.xpath("//div/h3[@data-test='error']");

    public void enterUsername(String username){
        driver.findElement(userNameXpath).clear();
        driver.findElement(userNameXpath).sendKeys(username);
    }
    public void enterPassword(String password){
        driver.findElement(passwordXpath).clear();
        driver.findElement(passwordXpath).sendKeys(password);
    }
    public void clickLogin(){
        driver.findElement(loginBtnXpath).click();
    }
    public String loginErrorDetails(){
        return getText(loginErrorXpath);
    }
}
